minifi-c2/minifi-c2-commons/src/main/java/org/apache/nifi/minifi/c2/model/OperationRequest.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.minifi.c2.model;
+
+import io.swagger.annotations.ApiModel;
+import io.swagger.annotations.ApiModelProperty;
+
+@ApiModel
+public class OperationRequest {
```

Comment:
This looks to be C2 server specific, is that true? If there is the need to push this further to a quorum of agents we will need a source identifier and then what's referred to as a request path.

Reply:
This is C2 server specific for now. I wanted to try to formulate a basic mechanism that someone could manually submit an operation request to the C2 server using it's rest api, which would then be queued in the C2 Server and eventually sent to the targeted MiNiFi agent over the C2 protocol (e.g., in a heartbeat response). I don't think there is a need to push this into the C2 protocol yet, but I could see how that would be useful eventually, so if you have an idea for how to do that I'd be happy to implement support in the C2 server.